include/eve/algo/container/detail/soa_storage2.hpp

```diff
@@ -0,0 +1,227 @@
+//==================================================================================================
+/*
+  EVE - Expressive Vector Engine
+  Copyright : EVE Contributors & Maintainers
+  SPDX-License-Identifier: MIT
+*/
+//==================================================================================================
+#pragma once
+
+#include <eve/algo/copy.hpp>
+#include <eve/algo/equal.hpp>
+#include <eve/algo/fill.hpp>
+#include <eve/algo/views/convert.hpp>
+#include <eve/algo/views/zip.hpp>
+#include <eve/concept/simd_allocator.hpp>
+#include <eve/detail/kumi.hpp>
+#include <eve/memory/aligned_allocator.hpp>
+#include <eve/module/core.hpp>
+#include <eve/product_type.hpp>
+#include <cstddef>
+#include <type_traits>
+#include <memory>
+
+namespace eve::algo::detail
+{
+  struct memory_helper
+  {
+    static void copy_all_aligned(std::byte const* src, std::byte* dst, std::size_t size)
+    {
+      auto aligned_src = eve::as_aligned(reinterpret_cast<std::uint8_t const*>(src), lane<64>);
+      eve::algo::copy ( as_range(aligned_src, aligned_src+size)
+                      , eve::as_aligned(reinterpret_cast<std::uint8_t*>(dst), lane<64>)
+                      );
+    }
+
+    template<typename Storage>
+    static void grow(Storage& s, std::size_t new_capacity, std::size_t old_size)
+    {
+      // Build new storage
+      Storage that{s.get_allocator(), new_capacity};
+
+      // Copy old data
+      using type  = typename Storage::value_type;
+      auto  b     = views::convert(s.data(), eve::as<type>{});
+      eve::algo::copy( as_range(b, b + old_size), that.data());
+
+      s = std::move(that);
+    }
+
+    template<typename Iterator, typename Value>
+    static void fill(Iterator begin, std::size_t first, std::size_t last, Value v)
+    {
+      eve::algo::fill(algo::as_range(begin + first, begin + last), v );
```

Comment:
Wait - no! Don't thrrow away alignment!

Reply:
The same remark, I sometimes use this fill with arbitrary first and last.